Meta/CMake/vcpkg/overlay-ports/angle/portfile.cmake

```diff
@@ -55,19 +53,29 @@ set(ANGLE_COMMIT_DATE "invalid-date")
 set(ANGLE_REVISION "${ANGLE_VERSION}")
 configure_file("${CMAKE_CURRENT_LIST_DIR}/angle_commit.h.in" "${SOURCE_PATH}/angle_commit.h" @ONLY)
 configure_file("${CMAKE_CURRENT_LIST_DIR}/angle_commit.h.in" "${SOURCE_PATH}/src/common/angle_commit.h" @ONLY)
+
+# Generate ANGLEShaderProgramVersion.h.in
+# FIXME: ANGLE's build system hashes the renderer files to determine the program version hash.
+#        For now, we'll just use the ANGLE commit hash.
+#        See: https://github.com/google/angle/commit/82826be01fcc4d02a637312f4df3ba97e74f7226#diff-81195814d06b98e6258a63901769078f42c522448b2847a33bd51e24ac9faef6
+set(ANGLE_PROGRAM_VERSION_HASH_SIZE 12)
+string(SUBSTRING "${ANGLE_COMMIT}" 0 ${ANGLE_PROGRAM_VERSION_HASH_SIZE} ANGLE_PROGRAM_VERSION)
+configure_file("${CMAKE_CURRENT_LIST_DIR}/ANGLEShaderProgramVersion.h.in" "${SOURCE_PATH}/ANGLEShaderProgramVersion.h" @ONLY)
+configure_file("${CMAKE_CURRENT_LIST_DIR}/ANGLEShaderProgramVersion.h.in" "${SOURCE_PATH}/src/common/ANGLEShaderProgramVersion.h" @ONLY)
+
 file(COPY "${CMAKE_CURRENT_LIST_DIR}/unofficial-angle-config.cmake" DESTINATION "${SOURCE_PATH}")
 
-set(ANGLE_WEBKIT_BUILDSYSTEM_COMMIT "bb1da00b9ba878d228a5e9834a0767dbca2fee43")
+set(ANGLE_WEBKIT_BUILDSYSTEM_COMMIT "624839dfc8c8b65c9ac45252fb1121ded39e3366")
 
 # Download WebKit gni-to-cmake.py conversion script
 vcpkg_download_distfile(GNI_TO_CMAKE_PY
     URLS "https://github.com/WebKit/WebKit/raw/${ANGLE_WEBKIT_BUILDSYSTEM_COMMIT}/Source/ThirdParty/ANGLE/gni-to-cmake.py"
     FILENAME "gni-to-cmake.py"
-    SHA512 9da35caf2db2e849d6cc85721ba0b77eee06b6f65a7c5314fb80483db4949b0b6e9bf4b2d4fc63613665629b24e9b052e03fb1451b09313d881297771a4f2736
+    SHA512 51ca45d4d2384d641b6672cb7cdfac200c58889b4b4cb83f1b04c1a0a2c9ab8b68f1c90d77763983684bcde674b073cfd85cfc160285332c0414d8ec6397601b
 )
 
 # Generate CMake files from GN / GNI files
-vcpkg_find_acquire_program(PYTHON3)
+x_vcpkg_get_python_packages(PYTHON_VERSION "3" OUT_PYTHON_VAR "PYTHON3" PACKAGES ply)
```

Comment:
I found a method in a few other ports (mesa, opencv, ...) that tells vcpkg to create a venv just for the specific port. Still not ideal, but better than adding a build dep on our end.